### PR TITLE
doc(framework): keras guide

### PR DESCRIPTION
**Pull request by larme**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

Feel free to tag members/contributors who can help review your PR.

### CODE REVIEW

File: docs/source/reference/frameworks/keras.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+=====
+Keras
+=====
+
+.. admonition:: About this page
+
+   This is an API reference for Keras in BentoML. Please refer to
+   :doc:`</frameworks/keras>` for more information about how to use
```
Comment (aarnphm) [inline docs/source/reference/frameworks/keras.rst:8]:
```suggestion
   :doc:`/frameworks/keras` for more information about how to use
```

File: docs/source/frameworks/keras.rst [hunk 2]
```diff
@@ -2,50 +2,142 @@
...
 
-   You can find more examples for **Keras** in our `gallery <https://github.com/bentoml/gallery>`_ repo.
+Using Runners
+-------------
 
-.. currentmodule:: bentoml.keras
+.. seealso::
 
-.. autofunction:: bentoml.keras.save_model
+   See :ref:`concepts/runner:Using Runners` doc for a general introduction to the Runner concept and its usage.
```
Comment (aarnphm) [inline docs/source/frameworks/keras.rst:136]:
```suggestion
   :ref:`concepts/runner:Using Runners`: a general introduction to the Runner concept and its usage.
```

File: docs/source/frameworks/keras.rst [hunk 3]
```diff
@@ -2,50 +2,142 @@
...
 
-.. autofunction:: bentoml.keras.load_model
 
-.. autofunction:: bentoml.keras.get
+``runner.predict.run`` is generally a drop-in replacement for
+``model.predict`` for executing the prediction in the model
+runner. When ``predict`` is the only prediction method exposed by
+runner model, you can just use ``runner.run`` instead of
+``runner.predict.run``
```
Comment (aarnphm) [inline docs/source/frameworks/keras.rst:143]:
```suggestion
runner. 

.. tip::

   When your Keras model only contains one inference signature (such as ``predict``), Runner also provides ``runner.run`` that is synonymous to ``runner.predict.run``.
```

File: docs/source/frameworks/keras.rst [hunk 4]
```diff
@@ -2,50 +2,142 @@
...
 
-.. code:: python
+   Both ``bentoml.keras`` and ``bentoml.tensorflow`` support Keras
+   model. ``bentoml.keras`` utilizing Keras' native model format,
+   which will give better development experience if user are more
+   familiar with Keras model. However, Keras' native model format is
+   not optimized for production inference. Using
+   ``bentoml.tensorflow`` is recommended in production
+   environment. You can read :doc:`bentoml.tensorflow documentation
+   </frameworks/tensorflow>` for more information.
```
Comment (aarnphm) [inline docs/source/frameworks/keras.rst:14]: Maybe for this part, bring the part in bentoml.tensorflow docs from jiangbo PR to here, and explain some of the performance deficit about using Keras.

File: docs/source/frameworks/keras.rst [hunk 5]
```diff
@@ -2,50 +2,142 @@
...
 
-    import bentoml
-    import tensorflow as tf
-    import tensorflow.keras as keras
+   You can also convert a Keras model to ONNX model and use
+   ``bentoml.onnx`` to serve production inference. Refer
+   :doc:`bentoml.onnx documentation </frameworks/onnx>` for more
+   information.
```
Comment (aarnphm) [inline docs/source/frameworks/keras.rst:19]: Worth noting to link to the official onnx to keras guide on this as well.

**Review by aarnphm — COMMENTED** [review]
Some comments

**Comment by ssheng** [comment]
Discussed with Larme. Larme will take one more iteration and mark this as ready for review.

**Review by ssheng — APPROVED** [review]

**Pull request merged**